### PR TITLE
[optimize](string) optimize constant empty string compare ( column='',  column!='')

### PR DESCRIPTION
**Pull request by ZhangYu0123**
# Proposed changes
Optimize constant empty string compare:
(1) When the constant empy string  '' (size is 0), we can compare offsets in SIMD directly.
 
````
q10: SELECT MobilePhoneModel, COUNT(DISTINCT UserID) AS u FROM hits WHERE MobilePhoneModel <> '' GROUP BY MobilePhoneModel ORDER BY u DESC LIMIT 10;
q11: SELECT MobilePhone, MobilePhoneModel, COUNT(DISTINCT UserID) AS u FROM hits WHERE MobilePhoneModel <> '' GROUP BY MobilePhone, MobilePhoneModel ORDER BY u DESC LIMIT 10;
q12: SELECT SearchPhrase, COUNT(*) AS c FROM hits WHERE SearchPhrase <> '' GROUP BY SearchPhrase ORDER BY c DESC LIMIT 10;
q13: SELECT SearchPhrase, COUNT(DISTINCT UserID) AS u FROM hits WHERE SearchPhrase <> '' GROUP BY SearchPhrase ORDER BY u DESC LIMIT 10;
q14: SELECT SearchEngineID, SearchPhrase, COUNT(*) AS c FROM hits WHERE SearchPhrase <> '' GROUP BY SearchEngineID, SearchPhrase ORDER BY c DESC LIMIT 10;
````

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by ZhangYu0123** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 33.68 seconds
 stream load tsv:          453 seconds loaded 74807831229 Bytes, about 157 MB/s
 stream load json:         21 seconds loaded 2358488459 Bytes, about 107 MB/s
 stream load orc:          72 seconds loaded 1101869774 Bytes, about 14 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230402221646_clickbench_pr_123839.html

**Comment by ZhangYu0123** [comment]
run p1

**Comment by ZhangYu0123** [comment]
run p1

File: be/src/vec/functions/functions_comparison.h [hunk 1]
```diff
@@ -211,15 +211,28 @@ struct StringEqualsImpl {
...
-            c[i] = positive == memequal_small_allow_overflow15(a_data.data() + prev_a_offset,
-                                                               a_size, b_data.data(), b_size);
-
-            prev_a_offset = a_offsets[i];
+        if (b_size == 0) {
+            auto* __restrict data = c.data();
+            auto* __restrict offsets = a_offsets.data();
+            if (positive) {
+                for (size_t i = 0; i < size; ++i) {
+                    data[i] = (offsets[i] == offsets[i - 1]);
```
Comment (HappenLee) [inline be/src/vec/functions/functions_comparison.h:219]: `data[i] = positive ? (offsets[i] == offsets[i - 1]) : (offsets[i] != offsets[i - 1]);`

The code can auto simd

**Comment by ZhangYu0123** [comment]
run buildall

File: be/src/vec/functions/functions_comparison.h (hunk 1 above)
Comment (HappenLee) [inline be/src/vec/functions/functions_comparison.h]: `data[i] = positive ? (offsets[i] == offsets[i - 1]) : (offsets[i] != offsets[i - 1]);`

The code can auto simd

Comment (ZhangYu0123) [reply be/src/vec/functions/functions_comparison.h]: done

**Comment by jacktengg** [comment]
LGTM

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**